README.md

```diff
@@ -15,83 +17,593 @@ yarn add gotrue-js
+
+Example usage:
+
+```
+auth.signup(email, password)
+  .then(response => console.log("Confirmation email sent", response))
+  .catch(error => console.log("It's an error",error));
+```
+
+Example response object:
+
+```
+{ id: 'example-id',
+  aud: '',
+  role: '',
+  email: 'example@example.com',
+  confirmation_sent_at: '2018-04-27T22:36:59.636416916Z',
+  app_metadata: { provider: 'email' },
+  user_metadata: null,
+  created_at: '2018-04-27T22:36:59.632133283Z',
+  updated_at: '2018-04-27T22:37:00.061039863Z' }
+```
+
+Also, make sure the `Registration preferences` under `Identity settings` in your Netlify dashboard are set to `Open`.
+
+[Screenshot]
+
+If the registration preferences is set to be `Invite only`, you'll get an error message as follows:
+`{code: 403, msg: 'Signups not allowed for this instance'}`
+
+### Confirm a new user signup
+
+Sends out an email to the user with a link to confirm their account.
+
+`auth.confirm(token)`
+
+A unique `confirmation_token` gets created when the user clicks on the link in the email.
+For security reason, the token is hidden from the browser via redirect.
+
+You can run `curl` to get the token in command line. E.g.,
+
+```
+{curl -I https://example.netlify.com?p=<example query>
+...
+Location: https://example.netlify.com/#confirmation_token=<example token>
+}
+```
+
+Example usage:
+
+`auth.confirm(token).then( response => console.log("Confirmation email sent", response), error => console.log("Error during signup: %o", error) );`
+
+Note that you'll need to test out this method in the browser instead of using a REPL like `node` as it requires usage of `localStorage`. Otherwise, you'll get an error message that includes `Error during confirmation ReferenceError: localStorage is not defined`.
+
+Example response object:
+
+```
+api: Object { apiURL: "https://example.netlify.com/.netlify/identity", \_sameOrigin: true, defaultHeaders: {} }
+app_metadata: Object { provider: "email" }
+aud: ""
+audience: undefined
+confirmation_sent_at: "2018-04-30T21:17:33Z"
+confirmed_at: "2018-04-30T21:21:31Z"
+created_at: "2018-04-30T21:01:38Z"
+email: "example@example.com"
+id: "example-id"
+role: ""
+token: Object { access_token: "example-token", token_type: "bearer", expires_in: 3600, … }
+updated_at: "2018-04-30T21:01:38Z"
+url: "https://example.netlify.com/.netlify/identity"
+user_metadata: Object { }
+```
+
+### Login a user
+
+Handles user login.
 
-auth.requestPasswordRecovery(email).then(
-  response => console.log("Recovery email sent"),
-  error => console.log("Error sending recovery mail: %o", error)
-);
+`auth.login(email, password)`
 
-auth.recover(token).then(
-  user => console.log("Logged in as %s", user.email),
-  error => console.log("Failed to verify recover token: %o", error)
-);
+Example usage:
 
+```
+auth
+  .login(email, password)
```

Comment:
I prefer when fluent/promise chains start on the same line.  We can align the subsequent calls though.